src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/settings/ProjectSettingsRemoteAjaxController.java

```diff
@@ -0,0 +1,136 @@
+
+/**
+ * Controller for managing settings for a remotely sync'd project
+ * asynchronously
+ */
+@RestController
+@RequestMapping("/ajax/remote-projects/{projectId}/settings")
+public class ProjectSettingsRemoteAjaxController {
+	private MessageSource messageSource;
+	private ProjectService projectService;
+	private ProjectRemoteService projectRemoteService;
+	private UserService userService;
+
+	@Autowired
+	public ProjectSettingsRemoteAjaxController(MessageSource messageSource, ProjectService projectService,
+			ProjectRemoteService projectRemoteService, UserService userService) {
+		this.messageSource = messageSource;
+		this.projectService = projectService;
+		this.projectRemoteService = projectRemoteService;
+		this.userService = userService;
+	}
+
+	/**
+	 * Update the project sync settings
+	 *
+	 * @param projectId                          the project id to update
+	 * @param remoteProjectSettingsUpdateRequest object which
+	 *                                           is used to update frequency, and sync user, as well as
+	 *                                           force sync for a project
+	 * @param principal                          The current logged in user
+	 * @param locale                             user's locale
+	 * @return result message if successful
+	 */
+	@PostMapping(value = "/sync")
+	public Map<String, String> updateProjectSyncSettings(@PathVariable Long projectId,
+			@RequestBody RemoteProjectSettingsUpdateRequest remoteProjectSettingsUpdateRequest,
+			Principal principal,
+			Locale locale) {
+
+		Project read = projectService.read(projectId);
+		RemoteStatus remoteStatus = read.getRemoteStatus();
+		Map<String, Object> updates = new HashMap<>();
+
+		String message = null;
+		String error = null;
+
+		if (remoteProjectSettingsUpdateRequest.getProjectSyncFrequency() != null) {
+			updates.put("syncFrequency", remoteProjectSettingsUpdateRequest.getProjectSyncFrequency());
+			message = messageSource.getMessage("project.settings.notifications.sync.frequencychange", new Object[] {remoteProjectSettingsUpdateRequest.getProjectSyncFrequency()}, locale);
+		}
+
+		if (remoteProjectSettingsUpdateRequest.getForceSync()) {
+			remoteStatus.setSyncStatus(RemoteStatus.SyncStatus.MARKED);
+			updates.put("remoteStatus", remoteStatus);
+			message = messageSource.getMessage("project.settings.notifications.sync.marked.for.sync", new Object[] {}, locale);
+		}
+
+		if (remoteProjectSettingsUpdateRequest.getChangeUser()) {
+			// ensure the user can read the project
+			try {
+				projectRemoteService.read(remoteStatus.getURL());
+
+				User user = userService.getUserByUsername(principal.getName());
+				remoteStatus.setReadBy(user);
+				updates.put("remoteStatus", remoteStatus);
+
+				message = messageSource.getMessage("project.settings.notifications.sync.userchange", new Object[] {},
+						locale);
+			} catch (Exception ex) {
+				error = messageSource.getMessage("project.settings.notifications.sync.userchange.error",
```

Comment:
Preface message keys returned from the server with `server.` that way we can keep track of where they are being used.  I also usually add the name of the component it is being used in: `server.MyComponent.userchange.error` for example.

Reply:
Updated in 3bd93720a5eb17ec1c86cc4c22cfa47f30d13312